src/components/ui/ClusterEventsToolbar.tsx

```diff
@@ -104,13 +124,18 @@ const ClusterEventsToolbar: React.FC<ClustersListToolbarProps> = ({
     const isChecked = event.target.checked;
 
     switch (value) {
-      case NO_HOSTS:
-        setFilters({
-          ...filters,
-          hosts: [], // clear all selection whenever clicked
-          clusterLevel: false,
-          orphanedHosts: false,
-        });
+      case SELECT_ALL:
+        if (filters.selectAll) {
+          setFilters({
+            ...filters,
+            clusterLevel: false,
+            orphanedHosts: false,
+            selectAll: false,
+            hosts: [],
+          });
+        } else {
+          setFilters(getInitialClusterEventsFilters(cluster));
```

Comment:
With this, the `Severity` and `Text search` are emptied when `Select all hosts` is checked. We should just mark all hosts here and keep other subfilters untouched.